Realm/Tests/DictionaryPropertyTests.m

```diff
@@ -1493,4 +1493,178 @@ - (void)testFrozenDictionarysDoNotUpdate {
     XCTAssertEqual(frozen.count, 2);
 }
 
+- (void)testAddEntriesFromDictionaryUnmanaged {
+    RLMDictionary<NSString *, StringObject *> *dict = [[DictionaryPropertyObject alloc] init].stringDictionary;
+    RLMAssertThrowsWithReason([dict addEntriesFromDictionary:@[@"string"]],
+                              @"Cannot add entries from object of class 'NSConstantArray'");
```

Comment:
This is failing because it is expecting `Cannot add entries from object of class '__NSSingleObjectArrayI`